client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -38,30 +37,55 @@ import { getEditorPostId } from 'state/ui/editor/selectors';
  */
 import './style.scss';
 
-class CalypsoifyIframe extends Component {
-	static propTypes = {
-		postId: PropTypes.number,
-		postType: PropTypes.string,
-		duplicatePostId: PropTypes.number,
-		pressThis: PropTypes.object,
-	};
+interface Props {
+	duplicatePostId: number;
+	postId: number;
+	postType: string;
+	pressThis: any;
+	siteAdminUrl: string;
+}
+
+type ConnectedProps = ReturnType< typeof mapStateToProps > & typeof mapDispatchToProps;
+
+interface State {
+	allowedTypes?: any;
+	classicBlockEditorId?: any;
+	editedPost?: any;
+	gallery?: any;
+	isIframeLoaded: boolean;
+	isMediaModalVisible: boolean;
+	isPreviewVisible: boolean;
+	multiple?: any;
+	postUrl?: string;
+	previewUrl: string;
+}
 
-	state = {
-		classicBlockEditorId: null,
+enum WindowActions {
+	Loaded = 'loaded',
+	ClassicBlockOpenMediaModel = 'classicBlockOpenMediaModal',
+}
+
+enum EditorActions {
+	GoToAllPosts = 'goToAllPosts',
+	OpenMediaModal = 'openMediaModal',
+	OpenRevisions = 'openRevisions',
+	PreviewPost = 'previewPost',
+	SetDraftId = 'draftIdSet',
+	TrashPost = 'trashPost',
+}
+
+class CalypsoifyIframe extends Component< Props & ConnectedProps, State > {
+	state: State = {
 		isMediaModalVisible: false,
 		isIframeLoaded: false,
 		isPreviewVisible: false,
 		previewUrl: 'about:blank',
-		postUrl: null,
-		editedPost: null,
 	};
 
-	constructor( props ) {
-		super( props );
-		this.iframeRef = React.createRef();
-		this.mediaSelectPort = null;
-		this.revisionsPort = null;
-	}
+	iframeRef: React.RefObject< HTMLIFrameElement > = React.createRef();
```

Comment:
this line isn't entirely necessary as written because I could have simply put `iframeRef = React.createRef()` but by typing the `HTMLIFrameElement` it not only specifies what it will refer to but opens up the further inference and autocomplete on those properties unique to the iframe, such as `contentWindow`, which may not be there in general `HTMLElement`s